backend/src/controller/chatController.ts

```diff
@@ -102,58 +156,44 @@ async function handleHigherLevelChat(
 		transformedMessage
 			? combineTransformedMessage(transformedMessage)
 			: message,
-		transformedMessage ? true : false,
-		sentEmails,
 		currentLevel
 	);
 
 	// run defence detection and chatGPT concurrently
-	const [defenceReport, openAiReply] = await Promise.all([
+	const [inputDefenceReport, openAiReply] = await Promise.all([
 		triggeredDefencesPromise,
 		openAiReplyPromise,
 	]);
 
-	// if input message is blocked, restore the original chat history and add user message (not as completion)
-	if (defenceReport.isBlocked) {
-		updatedChatHistory = [
-			...updatedChatHistory,
-			{
-				completion: null,
-				chatMessageType: CHAT_MESSAGE_TYPE.USER,
-				infoMessage: message,
-			},
-		];
-		updatedChatResponse = {
-			...updatedChatResponse,
-			defenceReport,
-		};
-	} else {
-		updatedChatHistory = openAiReply.chatHistory;
+	updatedChatResponse.openAIErrorMessage =
+		openAiReply.chatResponse.openAIErrorMessage;
 
-		// combine alerted and triggered defences
-		const combinedAlertedDefences = [
-			...defenceReport.alertedDefences,
-			...openAiReply.chatResponse.defenceReport.alertedDefences,
-		];
-		const combinedTriggeredDefences = [
-			...defenceReport.triggeredDefences,
-			...openAiReply.chatResponse.defenceReport.triggeredDefences,
-		];
-		// combine blocked
-
-		updatedChatResponse = {
-			...updatedChatResponse,
-			reply: openAiReply.chatResponse.completion?.content?.toString() ?? '',
-			wonLevel: openAiReply.chatResponse.wonLevel,
-			openAIErrorMessage: openAiReply.chatResponse.openAIErrorMessage,
-			defenceReport: {
-				...defenceReport,
-				alertedDefences: combinedAlertedDefences,
-				triggeredDefences: combinedTriggeredDefences,
-				isBlocked: openAiReply.chatResponse.defenceReport.isBlocked,
-				blockedReason: openAiReply.chatResponse.defenceReport.blockedReason,
-			},
-		};
+	const botReply = openAiReply.chatResponse.completion?.content?.toString();
+	const outputDefenceReport = botReply
+		? detectTriggeredOutputDefences(botReply, defences)
+		: null;
```

Comment:
This is where we're now looking for triggered defences on the bot output. This used to be done in openai.ts.